notebooks/segmented_model_demo/cat_encoder.py

```diff
@@ -31,9 +31,9 @@ def __call__(self, x):
     def transformed_features(self):
         return self._transformed_column_names
 
-    def cat_indexes_of_feature(self, feature: str) -> List[int]:
-        result = []
+    def cat_indexes_of_feature(self, feature: str) -> Dict[str,Dict[str,Any]]:
+        result = {}
         for idx, feat in enumerate(self._transformed_column_names):
             if feat.startswith(f"{feature}_"):
-                result.append(idx)
+                result[feat] = {'idx':idx, 'model': None}
```

Comment:
I don;t think you should return a dict with a concept of `model` from this class because that is assuming a lter usage that may only be true for this context prefer to keep CatEncoder generic and we'll back-migrate the added capability to the one we use in other examples too).
Also I think the keys should the feature *value* not the encoded one-hot-col name (so the substring *following* `{feature}_`).  Hence in this case I'd expect a dict like:
```
{
    "SVM": 62,
    "logistic": 63
}
```